### PR TITLE
Fix validator set and batch storage

### PR DESCRIPTION
**Pull request by goran-ethernal**
This PR fixes storing of signatures and bitmaps of signed validator sets and batches on `BridgeStorage` contract.

### CODE REVIEW

**Review by dusannosovic-ethernal — APPROVED** [review]

File: contracts/blade/BridgeStorage.sol [hunk 1]
```diff
@@ -26,9 +59,14 @@ contract BridgeStorage is ValidatorSetStorage {
         bytes memory hash = abi.encode(keccak256(abi.encode(batch)));
```
Comment (Stefan-Ethernal) [inline contracts/blade/BridgeStorage.sol:61]: Missing the @param doc for the `signature` and `bitmap` parameters.

File: contracts/blade/BridgeStorage.sol [hunk 2]
```diff
@@ -4,13 +4,46 @@ pragma solidity ^0.8.19;
...
+    event NewBatch(uint256 indexed id);
+    event NewValidatorSetStored(uint256 indexed id);
+
+    /**
+     * @notice commits new validator set
+     * @param newValidatorSet new validator set
+     * @param signature aggregated signature of validators that signed the new validator set
+     * @param bitmap bitmap of which validators signed the message
+     */
+    function commitValidatorSet(
```
Comment (Stefan-Ethernal) [inline contracts/blade/BridgeStorage.sol:25]:
```suggestion
    function commitValidatorSetAndSignatures(
```

File: contracts/blade/ValidatorSetStorage.sol [hunk 3]
```diff
@@ -136,4 +128,20 @@ contract ValidatorSetStorage is IValidatorSetStorage, Initializable, System {
         // Get the value of the bit at the given 'index' in a byte.
         return uint8(bitmap[byteNumber]) & (1 << bitNumber) > 0;
     }
+
+    function _commitValidatorSet(
```
Comment (Stefan-Ethernal) [inline contracts/blade/ValidatorSetStorage.sol:140]: Let's add some comment, e.g. (probably can be shorten)


```suggestion
    /**
    * @notice Commits a new validator set by verifying its signature and setting it in the storage.
    * @dev This function requires that the new validator set is non-empty and verifies its signature 
    *      before updating the validator set.
    * 
    * @param newValidatorSet The array of validators to be set as the new validator set. 
    *                        It must be non-empty, and each validator is represented as a `Validator` struct.
    * @param signature The aggregated BLS signature over the new validator set. It is an array with two elements 
    *                  representing the signature in a G1 point.
    * @param bitmap The bitmap representing which validators signed the new validator set. It is a byte array where
    *               each bit indicates whether a particular validator in the previous set has signed the new set.
    * 
    * Emits a `NewValidatorSet` event after successfully setting the new validator set.
    */
    function _commitValidatorSet(
```

Comment (goran-ethernal) [reply contracts/blade/ValidatorSetStorage.sol:140]: Fix: https://github.com/Ethernal-Tech/blade-contracts/pull/36/commits/d5c6a9e32d929e329c727dcb8d6c2295bc589c4f

File: contracts/blade/BridgeStorage.sol (hunk 1 above)
Comment (goran-ethernal) [reply contracts/blade/BridgeStorage.sol:61]: Fix: https://github.com/Ethernal-Tech/blade-contracts/pull/36/commits/d5c6a9e32d929e329c727dcb8d6c2295bc589c4f

**Review by Stefan-Ethernal — APPROVED** [review]

**Pull request merged**